README.md

```diff
@@ -145,6 +145,18 @@ npm run dev:site
 npm run test
 ```
 
+### Tooling
+
+Git hooks with husky will lint the code before commit and run tests before push. So before pushing, running the following command at least once is always needed.
+
+```bash
+# Launch the documentation website
+npm run bootstrap
+
+# before push
+npm run dev:site
+```
```

Comment:
Hm, I'm curious about this. Basically, `npm run bootstrap` installs all the deps in the monorepo and should be executed at first, just after the `git clone`, but no after that. And `npm run dev:site` is launching the dev server, like a `npm start` or `npm run dev` with respectively CRA or Next.js app.
I don't understand why theses cmds help you to pass git hook requirements 🧐

Instead, consider these scripts: 

```sh
# build usehook-ts library (exec types check internally)
npm run build:lib

# Exec both all linter, types check and unit-tests
npm run test
```

Normally, these two commands will catch major codebase errors and then, allow a `git push`.

Reply:
I'll try to delete the repo and reinstall and see if it works on my side. Sorry for the delay.

Reply:
here's the result of my tests:

1. husky pre-commit only works after `npm run bootstrap` so small changes can be made without initialization.
2. when a hook doesn't import from `'usehooks-ts'` the lint-staged won't need `build:lib`
3. publishing a branch will trigger `pre-push` and `npm run test` so the `npm run build:lib` is needed.

To sum up: after any code change with `npm run bootstrap`, the `build:lib` command is needed and the `npm run test` is not that much needed (if we use CLI to push, or haven't `npm run bootstrap`.) 

I'm thinking to merge this part to `### Create a new hook` or rename it to `### Update a existed hook`. What's your preference?

Reply:
Your tests are ok, but I disagree with the shape.
We should recommend script for that it is intended to, I mean, `npm run bootstrap` isn't made for passing git-hooks and so, shouldn't describe in the doc as a requirement step to push. Instead, It should be run once at first to initiate the project locally. (We need to follow this rule for others script tough).

So, yes, you could create a `### Update an existing hook` block. 

```sh
# Install the whole workspace
npm run bootstrap

# Compile the library
npm run build:lib

# Exec Typescript, linter and unit tests
npm run test
```

These scripts aren't made to pass the git-hooks, the git-hooks are made to verify that the build steps are respected 🤣

Otherwise, it's a good plus to the doc, thanks 😊